logfile_iterator_for_video_device/src/main.c

```diff
@@ -0,0 +1,431 @@
+            if(strncmp(outputformat, "bmp", 3) == 0)
+            {
+                context->output_format = OUTPUT_BMP;
+            }
+            else if(strncmp(outputformat, "ppm", 3) != 0)
+            {
+                psync_log_message(
+                    LOG_LEVEL_ERROR,
+                    "unsupported output format, supported formats are bmp and ppm");
+                ret = DTC_USAGE;
+                poptPrintUsage(opt_ctx, stderr, 0);
+            }
+        }
+    }
+
+    if(ret == DTC_NONE)
+    {
+        if(logfilepath_set == 0)
+        {
+            psync_log_message(
+                LOG_LEVEL_ERROR,
+                "path to logfile required");
+            poptPrintUsage(opt_ctx, stderr, 0);
+        }
+        else
+        {
+            int print_ret = 0;
+
+            print_ret = snprintf(
+                context->logfile_path,
+                sizeof(context->logfile_path),
+                "%s",
+                logfilepath);
+
+                if(print_ret < 0)
+            {
+                psync_log_message(
+                    LOG_LEVEL_ERROR,
+                    "error setting logfile path! snprintf returned %d",
+                    print_ret);
+                ret = DTC_DATAERR;
+            }
+        }
+    }
+
+    if(ret == DTC_NONE)
+    {
+        if(outdir_set != 0)
+        {
+            int print_ret = 0;
+            print_ret = snprintf(
+                context->output_dir,
+                sizeof(context->output_dir) - 1,
+                "%s", outdir);
+
+            // Minimal effort path cleanup
+            size_t len = strlen(context->output_dir);
+
+            if(context->output_dir[len-1] == '/')
+            {
+                context->output_dir[len-1] = '\0';
+            }
+
+            if(print_ret < 0)
+            {
+                psync_log_message(
+                    LOG_LEVEL_ERROR,
+                    "error setting outdir! snprintf returned %d",
+                    print_ret);
+                ret = DTC_DATAERR;
+            }
+        }
+    }
+
+    return ret;
+}
+
+int main(int argc, char **argv)
+{
+    int ret = DTC_NONE;
+    context_s context;
+    memset(&context, 0, sizeof(context));
+
+    ret = init_context(&context, NULL);
+
+    if(ret == DTC_NONE)
+    {
+        ret = parse_options(argc, argv, &context);
+    }
+
+    if(ret == DTC_NONE)
+    {
+        psync_log_message(
+            LOG_LEVEL_INFO,
+            "writing image files to %s", context.output_dir);
+    }
+
+    if(ret == DTC_NONE)
+    {
+        // init core API
+        ret = psync_init(
+            NODE_NAME,
+            PSYNC_NODE_TYPE_API_USER,
+            PSYNC_DEFAULT_DOMAIN,
+            PSYNC_SDF_ID_INVALID,
+            PSYNC_INIT_FLAG_STDOUT_LOGGING,
+            &context.node_ref);
+
+            if(ret != DTC_NONE)
+            {
+                psync_log_message(
+                        LOG_LEVEL_ERROR,
+                        "main -- psync_init - ret: %d",
```

Comment:
Do we need to have the 'main --' part? Some of the other messages don't have that format.